### PR TITLE
FOCUS #242: Creating SKU ID based on work done by @arrarama

### PR DESCRIPTION
**Pull request by udam-f2**
Creating SKU ID (continuation from https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/239)
Author: arrarama

### CODE REVIEW

File: supporting_content/dimensions/skuid.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example provider mappings 
+
+Current column mappings found in available data sets:
+
+| Provider     | Data set                    | Column          |
+| ------------ | --------------------------- | --------------- |
+| AWS          | CUR                         | product/sku     |
+| Azure        | Cost details export or API  | PartNumber      |
+| GCP          | Big Query Export            | sku.id          |
```
Comment (salpaysenturus) [inline supporting_content/dimensions/skuid.md:11]:
```suggestion
| GCP          | BigQuery Billing Export            | sku.id          |
```

File: supporting_content/dimensions/skuid.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example scenarios for current provider data
+
+Current values observed in billing data for various scenarios:
+
+| Provider     | Scenario                   | Pattern          |
+| ------------ | -------------------------- | ---------------- |
+| AWS          | CUR                        | FFNT87MQSCR328W6 |
+| Azure        | Cost Details export or API |  ABC-12345       |
+| GCP          | Big Query                  | F1ED-0732-5BDA   |
```
Comment (salpaysenturus) [inline supporting_content/dimensions/skuid.md:24]:
```suggestion
| GCP          | BigQuery Billing Export                  | F1ED-0732-5BDA   |
```

**Review by salpaysenturus — APPROVED** [review]
LGTM

File: specification/dimensions/skuid.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# SKU ID
+
+A SKU ID is a unique identifier that defines the SKU associated with the charge. SKU ID can be referenced on a price list published by a provider to look up detailed information about the SKU and the identified unit price associated with the SKU. The composition of the detailed information associated with the SKU may differ between providers.
+
+The SkuId column MUST be present in the billing data. This column MUST be of type String. The SkuId MUST NOT be null when Charge Type is 'Purchase' or 'Usage'. The SkuId SHOULD NOT be null when Adjustment Type is 'Refund'.
```
Comment (salpaysenturus) [inline specification/dimensions/skuid.md:5]: Per the weekly FOCUS call today: 

```suggestion
The SkuId column MUST be present in the billing data. This column MUST be of type String. The SkuId MUST NOT be null when Charge Type is 'Purchase' or 'Usage'. The SkuId MUST be null when Adjustment Type is 'Tax'.  The SkuId SHOULD NOT be null when Adjustment Type is 'Refund'.
```

File: supporting_content/dimensions/skuid.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+Current column mappings found in available data sets:
+
+| Provider     | Data set                    | Column          |
+| ------------ | --------------------------- | --------------- |
+| AWS          | CUR                         | product/sku     |
+| Azure        | Cost details export or API  | PartNumber      |
+| GCP          | BigQuery Billing Export     | sku.id          |
+| OCI          | Cost Reports                | cost/productSku |
+|              |                             |                 |
```
Comment (salpaysenturus) [inline supporting_content/dimensions/skuid.md:13]: Remove blank line

```suggestion

```

File: supporting_content/dimensions/skuid.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+Current values observed in billing data for various scenarios:
+
+| Provider     | Scenario                   | Pattern          |
+| ------------ | -------------------------- | ---------------- |
+| AWS          | CUR                        | FFNT87MQSCR328W6 |
+| Azure        | Cost Details export or API | ABC-12345       |
+| GCP          | BigQuery Billing Export    | F1ED-0732-5BDA   |
+| OCI          | Cost Reports               | B91962           |
+|              |                            |                  |
```
Comment (salpaysenturus) [inline supporting_content/dimensions/skuid.md:25]: Remove blank line

```suggestion

```

File: specification/dimensions/skuid.md (hunk 3 above)
Comment (joshk0) [reply specification/dimensions/skuid.md:5]: is it a MUST NOT for 'Refund'?

Comment (arrarama) [reply specification/dimensions/skuid.md]: Logic moved to SkuPriceId.

**Review by cnharris10 — APPROVED** [review]

**Review by mike-finopsorg — APPROVED** [review]

**Review by ljadvey — APPROVED** [review]

**Comment by shruthinambi** [comment]
A small correction in this - for GCP the dataset should be BigQuery Pricing export. The source for SKU and pricing related fields is the pricing export.

**Review by ijurica — APPROVED** [review]

File: supporting_content/dimensions/skuid.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+## Discussion / Scratch space
+
+References
+
+AWS - (<https://docs.aws.amazon.com/cur/latest/userguide/product-columns.html#product-details-P>)
+
+Azure - (<https://learn.microsoft.com/en-us/azure/cost-management-billing/automate/understand-usage-details-fields>)
+
+Big Query - (<https://cloud.google.com/billing/docs/how-to/export-data-bigquery-tables/detailed-usage>)
```
Comment (udam-f2) [inline supporting_content/dimensions/skuid.md:33]:
```suggestion
GCP - (<https://cloud.google.com/billing/docs/how-to/export-data-bigquery-tables/detailed-usage>)
(also, SKU information can be found via the BigQuery Pricing Export)
```

**Comment by udam-f2** [comment]
> A small correction in this - for GCP the dataset should be BigQuery Pricing export. The source for SKU and pricing related fields is the pricing export.

Hi! in the supporting content, we're referencing where to find the sku id information in the billing data set. Added a note in the discussion area about the pricing export. Thanks

**Pull request merged**